dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehavior.cs

```diff
@@ -83,11 +99,12 @@ public static FunctionChoiceBehavior None(IEnumerable<KernelFunction>? functions
     /// <summary>
     /// Returns functions AI connector should provide to the AI model.
     /// </summary>
+    /// <param name="functionFQNs">Functions provided as fully qualified names.</param>
     /// <param name="functions">Functions provided as instances of <see cref="KernelFunction"/>.</param>
     /// <param name="kernel">The <see cref="Kernel"/> to be used for function calling.</param>
     /// <param name="autoInvoke">Indicates whether the functions should be automatically invoked by the AI connector.</param>
     /// <returns>The configuration.</returns>
-    public IReadOnlyList<KernelFunction>? GetFunctions(IEnumerable<KernelFunction>? functions, Kernel? kernel, bool autoInvoke)
+    public IReadOnlyList<KernelFunction>? GetFunctions(IList<string>? functionFQNs, IEnumerable<KernelFunction>? functions, Kernel? kernel, bool autoInvoke)
```

Comment:
I understand you. I did not like this initially as well, but later, comparing it to other options, it appeared not as bad as it seemed. The method accepts different function sources and returns a final list of functions. There is no need to have 'if' logic in each subclass; all the code, relatively small, resides in one method rather than being scattered among a few. Taking into account that it's protected and the class is internal (its constructor is internal), we can change it later without the risk of breaking anything if needed.

Reply:
fixed